lib/grandpa/message_handler.go

```diff
@@ -87,18 +96,58 @@ func (h *MessageHandler) handleNeighbourMessage(msg *NeighbourMessage) error {
 
 	// TODO; determine if there is some reason we don't receive justifications in responses near the head (usually),
 	// and remove the following code if it's fixed. (#1815)
-	head, err := h.blockState.BestBlockNumber()
+	// head, err := h.blockState.BestBlockNumber()
+	// if err != nil {
+	// 	return err
+	// }
+
+	// TODO: Why are we ignoring these? Isn't
+	// msg.Number likely to be higher if we are lagging behind?
+	// ignore neighbour messages that are above our head
+	// if int64(msg.Number) > head.Int64() {
+	// 	return nil
+	// }
```

Comment:
we shouldn't send a catch up request for blocks we haven't synced yet as we won't be able to process them

Reply:
we also receive neighbour messages each time a new block is finalized, so we get them very often

Reply:
uncommented it